apps/remote_control/lib/lexical/remote_control/code_mod/rename/file.ex

```diff
@@ -0,0 +1,154 @@
+defmodule Lexical.RemoteControl.CodeMod.Rename.File do
+  alias Lexical.Ast
+  alias Lexical.Document
+  alias Lexical.ProcessCache
+  alias Lexical.Project
+  alias Lexical.RemoteControl
+  alias Lexical.RemoteControl.CodeIntelligence.Entity
+  alias Lexical.RemoteControl.Search.Indexer
+  alias Lexical.RemoteControl.Search.Indexer.Entry
+
+  @spec maybe_rename(Document.t(), Entry.t(), String.t()) :: Document.Changes.rename_file()
+  def maybe_rename(%Document{} = document, %Entry{} = entry, new_suffix) do
+    if root_module?(entry, document) do
+      rename_file(document, entry, new_suffix)
+    end
+  end
+
+  defp root_module?(entry, document) do
+    entries =
+      ProcessCache.trans("#{document.uri}-entries", 50, fn ->
```

Comment:
This is a cache set up before the rename results are returned, independent of the editor, mainly to prevent repeatedly indexing a file. 50 milliseconds should not be a problem.

Reply:
This is why i was asking if the rename process always exists, it might make sense to just not time this out and let the process exiting clear the cache.

Reply:
Please note that the process for `Command.Rename` is not the same as this process. The events here occur before `Command.Rename`.